bot/cogs/help_channels.py

```diff
@@ -0,0 +1,672 @@
+
+            if not channel:
+                log.info("Couldn't create a candidate channel; waiting to get one from the queue.")
+                await self.notify()
+                channel = await self.wait_for_dormant_channel()
+
+        return channel
+
+    @staticmethod
+    def get_position(channel: discord.TextChannel, destination: discord.CategoryChannel) -> int:
+        """Return alphabetical position for `channel` if moved to `destination`."""
+        log.trace(f"Getting alphabetical position for #{channel} ({channel.id}).")
+
+        # If the destination category is empty, use the first position
+        if not destination.channels:
+            position = 1
+        else:
+            # Make a sorted list of channel names for bisect.
+            channel_names = [c.name for c in destination.channels]
+
+            # Get location which would maintain sorted order if channel was inserted into the list.
+            rank = bisect.bisect(channel_names, channel.name)
+
+            if rank == len(destination.channels):
+                # Channel should be moved to the end of the category.
+                position = destination.channels[-1].position + 1
+            else:
+                # Channel should be moved to the position of its alphabetical successor.
+                position = destination.channels[rank].position
+
+        log.trace(
+            f"Position of #{channel} ({channel.id}) in {destination.name} will be {position} "
+            f"(was {channel.position})."
+        )
+
+        return position
+
+    @staticmethod
+    def get_category_channels(category: discord.CategoryChannel) -> t.Iterable[discord.TextChannel]:
+        """Yield the text channels of the `category` in an unsorted manner."""
+        log.trace(f"Getting text channels in the category '{category}' ({category.id}).")
+
+        # This is faster than using category.channels because the latter sorts them.
+        for channel in category.guild.channels:
+            if channel.category_id == category.id and isinstance(channel, discord.TextChannel):
+                yield channel
+
+    @staticmethod
+    def get_names() -> t.List[str]:
+        """
+        Return a truncated list of prefixed element names.
+
+        The amount of names is configured with `HelpChannels.max_total_channels`.
+        The prefix is configured with `HelpChannels.name_prefix`.
+        """
+        count = constants.HelpChannels.max_total_channels
+        prefix = constants.HelpChannels.name_prefix
+
+        log.trace(f"Getting the first {count} element names from JSON.")
+
+        with Path("bot/resources/elements.json").open(encoding="utf-8") as elements_file:
+            all_names = json.load(elements_file)
+
+        if prefix:
+            return [prefix + name for name in all_names[:count]]
+        else:
+            return all_names[:count]
+
+    def get_used_names(self) -> t.Set[str]:
+        """Return channels names which are already being used."""
```

Comment:
Intentional typo?
```suggestion
        """Return channel names which are already being used."""
```

Reply:
The typo I mentioned in the meeting was fixed in 00c9f092a0c056b21e012fdadd807d0410a3ca09 😄